src/router_core/transfer.c

```diff
@@ -412,14 +412,41 @@ static void qdr_delete_delivery_internal_CT(qdr_core_t *core, qdr_delivery_t *de
     free_qdr_delivery_t(delivery);
 }
 
+static bool qdr_delivery_has_peer_CT(qdr_delivery_t *dlv)
+{
+    if (!dlv->peer && DEQ_SIZE(dlv->peers) == 0)
```

Comment:
You could do this more elegantly like this:
    return dlv->peer || DEQ_SIZE(dlv->peers) > 0;